src/main/java/br/ufes/inf/nemo/vpzy/engine/FreeMarkerEngine.java

```diff
@@ -0,0 +1,63 @@
+package br.ufes.inf.nemo.vpzy.engine;
+
+import br.ufes.inf.nemo.vpzy.engine.models.entity.ClassModel;
+import br.ufes.inf.nemo.vpzy.logging.Logger;
+import freemarker.cache.FileTemplateLoader;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.logging.Level;
+
+public class FreeMarkerEngine {
+    private final Configuration cfg;
+
+    private String outputDirectory = "/src/main/java";
```

Comment:
Fixed: The base path for Java projects has been removed. However, a way to define the file extension based on the diagram couldn't be found. It is advised to keep it as ".java" for this issue. In the next iteration, when adding more templates, this can be addressed and necessary adjustments can be made.